src/Workspaces/Core/Portable/CodeGeneration/Symbols/CodeGenerationAbstractMethodSymbol.cs

```diff
@@ -32,6 +32,7 @@ public virtual ImmutableArray<AttributeData> GetReturnTypeAttributes()
         }
 
         public abstract int Arity { get; }
+        public abstract System.Reflection.MethodImplAttributes ImplementationAttributes { get; }
```

Comment:
This *might* be incorrect. Should I attempt to retrieve it through the `attributes` passed to the ctor of this abstract class? Hope someone can confirm.

Reply:
This stuff is used by the IDE when generating methods. I don't believe they're going to care about this API, so I don't think it actually matters. @jasonmalinowski to confirm what they'd like the impl strategy to be though.

Reply:
Agreed, we're be unlikely to care. These symbols are created when we we're using some of our code generation APIs that generate syntax from IMethodSymbols. Unless that code is also being updated to actually _generate_ this method, nothing is going to look at any implementation here.

So just to clarify, what's the behavior of symbols you get from the compiler?

**From Metadata:** ImplementationAttributes is correct, GetAttributes returns empty (since it's baked into the flags...?)
**From Source:** GetAttributes returns symbols, ImplementationAttributes is ...?

Reply:
> So just to clarify, what's the behavior of symbols you get from the compiler?

@jasonmalinowski `ImplementationAttributes` will be correct in all cases. In source cases, you'd get the attributes in `GetAttributes` as well.